packages/lisk-transactions/test/4_multisignature_transaction.ts

```diff
@@ -407,4 +409,99 @@ describe('Multisignature transaction class', () => {
 			expect(errors).to.be.empty;
 		});
 	});
+
+	describe('#addMultisignature', () => {
+		let membersSignatures: Array<SignatureObject>;
+		let multisigTrs: MultisignatureTransaction;
+
+		beforeEach(() => {
+			multisigTrs = new MultisignatureTransaction(
+				validMultisignatureRegistrationTransactionNoSigs,
+			);
+
+			membersSignatures = [
+				{
+					publicKey:
+						'142d1f24e17d3c90b0f2abdf49c2b14b02782e49b2ecfe85462ed12f654d60df',
+					signature:
+						'd1b78f5eb35b4e1de7f740d2f62f0e2acab24c5b446719cc70601319f4a3666fbcda7980e5d9c6ff3bfa8b54ee383eed5531723e0f1748d0c84b7a229759b000',
+					transactionId: multisigTrs.id,
+				},
+				{
+					publicKey:
+						'bb7ef62be03d5c195a132efe82796420abae04638cd3f6321532a5d33031b30c',
+					signature:
+						'aa956854dae10792ba9006e4dddd0d7e370af4645df8708d1b10f088304320f0e7f2427d883755a997c07b53978c4d8cf56b95e0f740d8d50ed8c1f2dc85180f',
+					transactionId: multisigTrs.id,
+				},
+			];
+		});
+
+		it('should add signature to transaction', async () => {
+			const { status } = multisigTrs.addMultisignature(
+				store,
+				membersSignatures[0],
+			);
+
+			expect(status).to.eql(Status.PENDING);
+			expect(multisigTrs.signatures).to.include(membersSignatures[0].signature);
+		});
+
+		it('should fail when valid signature already present and sent again', async () => {
+			const { status: arrangeStatus } = multisigTrs.addMultisignature(
+				store,
+				membersSignatures[0],
+			);
+
+			const { status, errors } = multisigTrs.addMultisignature(
+				store,
+				membersSignatures[0],
+			);
+			const expectedError = 'Encountered duplicate signature in transaction';
+
+			expect(arrangeStatus).to.eql(Status.PENDING);
+			expect(status).to.eql(Status.FAIL);
+			expect(errors[0].message).to.be.eql(expectedError);
+			expect(multisigTrs.signatures).to.include(membersSignatures[0].signature);
+		});
+
+		it('should fail to add invalid signature to transaction', async () => {
+			const { status, errors } = multisigTrs.addMultisignature(store, {
+				transactionId: multisigTrs.id,
+				publicKey:
+					'bb7ef62be03d5c195a132efe82796420abae04638cd3f6321532a5d33031b30c',
+				signature:
+					'eeee799c2d30d2be6e7b70aa29b57f9b1d6f2801d3fccf5c99623ffe45526104b1f0652c2cb586c7ae201d2557d8041b41b60154f079180bb9b85f8d06b3010c',
+			});
+
+			const expectedError =
+				'Failed to add signature eeee799c2d30d2be6e7b70aa29b57f9b1d6f2801d3fccf5c99623ffe45526104b1f0652c2cb586c7ae201d2557d8041b41b60154f079180bb9b85f8d06b3010c.';
+
+			expect(status).to.eql(Status.FAIL);
+			expect(errors[0].message).to.be.eql(expectedError);
```

Comment:
can you check in this test that the status of the Multisignature transaction is still PENDING and has not changed to FAIL?

Reply:
@SargeKhan added a new test for this as `_multisignatureStatus` is 0 until at least one valid signature is added. Should I create an issue so when a multi-signature transaction arrives without signatures its `_multisignatureStatus` is set to 2 -Pending-?

Reply:
` when a multi-signature transaction arrives without signatures its _multisignatureStatus is set to 2 -Pending-?`
in case of type 4, i think we should set the status to at least pending because we know it is multi-sig. Lets create an issue for that

Reply:
Created https://github.com/LiskHQ/lisk-elements/issues/1192